controller/volume_controller.go

```diff
@@ -1405,6 +1410,19 @@ func (c *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[strin
 	return c.checkAndFinishVolumeRestore(v, e, rs)
 }
 
+func (c *VolumeController) checkVolumeMountFileSystemStatus(v *longhorn.Volume, e *longhorn.Engine) (err error) {
+	log := getLoggerForVolume(c.logger, v)
+	if v.Status.State == longhorn.VolumeStateAttached && e.Status.CurrentState == longhorn.InstanceStateRunning {
+		if e.Status.FilesystemReadOnly {
```

Comment:
Why not just check `e.Status.HealthConditions[imtypes.EngineConditionFilesystemReadOnly]`?

Reply:
I agree with this change so we don't have to copy the status from `e.Status.HealthConditions` to other status field
This is originally from @derekbit 's advice 
cc @derekbit WDYT?

Reply:
Yeah, we don't need ` e.Status.FilesystemReadOnly` because you've already introduced `e.Status.HealthConditions`.

BTW, as discussion with @ChanYiLin, we can add FilesystemReadOnly condition in the `conditions` and no need to have a new `healthConditions`.